tracing-core/src/dispatcher.rs

```diff
@@ -221,10 +222,26 @@ pub fn with_default<T>(dispatcher: &Dispatch, f: impl FnOnce() -> T) -> T {
     // prior default. Using this (rather than simply resetting after calling
     // `f`) ensures that we always reset to the prior dispatcher even if `f`
     // panics.
-    let _guard = State::set_default(dispatcher.clone());
+    let _guard = set_default(dispatcher);
     f()
 }
 
+/// Sets the dispatch as the default dispatch for the duration of the lifetime
+/// of the returned DefaultGuard
+///
+/// TODO: Is this still valid?
```

Comment:
?

Reply:
Oh this slipped past me. I wasn't sure if the note following this line was valid.

Reply:
It is still valid. Just remove the TODO and we're good!
```suggestion
```